### PR TITLE
scenario comaparision, spatial configuration and system operations

### PR DESCRIPTION
**Pull request by drifter089**
- using conda env to run the app
- configuration options using config file

### CODE REVIEW

File: app/pages/1_Scenario_comparison.py [hunk 1]
```diff
@@ -1,6 +1,63 @@
...
 
 
-st.title("Scenario comparison")
 
-st.write("Scenario comparison plots will be added here.")
\ No newline at end of file
+def main():
+    st.title("Scenario comparison")
+
+    network_map = tools.get_network_map("pypsa-earth")
```
Comment (pz-max) [inline app/pages/1_Scenario_comparison.py]: Any reason why `pypsa-earth` is hard coded?

Comment (pz-max) [reply app/pages/1_Scenario_comparison.py]: There are many other places where this is the case

File: README.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ git clone https://github.com/pypsa-meets-earth/pypsa-earth-lit.git
 
 # depending on what you need. Recommended order.
 make -f ./pypsa-earth-lit/Makefile get_data
```
Comment (pz-max) [inline README.md:17]: I think this function will currently install `pypsa-earth` results in this way:
```
tauritron-project/pypsa-earth
tauritron-project/pypsa-earth-lit
```
However, as `pypsa-earth` is hard-coded in later script,
it currently runs only when the pypsa-earth data is in 
`pypsa-earth-lit/pypsa-earth`

Suggestion. Don't hard code `pypsa-earth` across scripts.
Keep if possible the original structure:
```
tauritron-project/pypsa-earth
tauritron-project/pypsa-earth-lit
```
and test if it works.

**Review by pz-max — CHANGES_REQUESTED** [review]

File: env.yaml
Comment (pz-max) [inline env.yaml:1]: changed the `env.yaml` such that it works on any OS

File: README.md (hunk 2 above)
Comment (drifter089) [reply README.md:17]: I have changed the pypsa earth path and tested it with the readme file it's working fine now.

File: app/pages/utils/spatial_pre_run.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    nLines["original_capacity"] = nLines.s_nom.clip(lower=1e-3)
+    nLines["max_capacity"] = nLines.s_nom_min.clip(lower=1e-3)
+
+    return nLines
+
+
+##### add data to these df and define them in config #####
+
+def get_spatial_values_df(pypsa_network,gpd_bus_regions):
+    # this df is a multiindex df with carrier and parameter as index and bus regions as columns
```
Comment (drifter089) [inline app/pages/utils/spatial_pre_run.py:81]: @pz-max add values in df here

File: app/pages/utils/spatial_pre_run.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    # add new row with index (carrier,parameter) and values as list of values for each bus region (in order)
+    # what ever parameters are added to indexes in this df should be added to config "spatial_parameters" along with nice names
+    base_df=_get_gen_df(pypsa_network,gpd_bus_regions)
+
+    return base_df
+
+
+def get_edges_df(pypsa_network):
+    #  making base df for edges with lines
```
Comment (drifter089) [inline app/pages/utils/spatial_pre_run.py:90]: @pz-max add data for lines/edges here

File: app/pages/utils/spatial_pre_run.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        merged_gen_df=carrier_df.merge(temp_name_series, left_on="bus", right_on="name", how="right")
+        # putting it in the param_bus_value_df 
+        for param in slected_cols_gen:
+            param_bus_value_df.loc[(carrier_in_unique, param)]=list(merged_gen_df[param])
+
+    param_bus_value_df.replace([np.inf, -np.inf,np.nan], 0, inplace=True)
+    return param_bus_value_df
+
+def _make_plot_lines_df(pypsa_network):
+    nLines=pypsa_network.lines.copy()
```
Comment (drifter089) [inline app/pages/utils/spatial_pre_run.py:69]: @pz-max update edge data (errors)

**Comment by pz-max** [comment]
I am currently reviewing this and worked on @drifter089  branch by applying this: https://stackoverflow.com/a/37686809

Code below needs to be adapted but give you an idea how to work on another branch @ekatef
```
git clone https://github.com/pz-max/PyPSA.git
cd PyPSA
git remote add AnasAlgarei https://github.com/AnasAlgarei/PyPSA
git fetch AnasAlgarei
git checkout -b load-example AnasAlgarei/load_shedding_shifting_example
```

**Comment by ekatef** [comment]
@pz-max thanks a lot for providing git-spells!
It worked :)

@drifter089 thanks a lot for the README and great presentation

**Pull request closed without merging**